src/plaid/bridges/huggingface_bridge.py

```diff
@@ -135,8 +142,65 @@ def plaid_generator_to_huggingface(
     return ds
 
 
+def huggingface_description_to_problem_definition(
+    description: dict,
+) -> ProblemDefinition:
+    """Converts a huggingface dataset description to a plaid problem definition.
+
+    Args:
+        description (dict): the description field of a huggingface dataset, containing the problem definition
+
+    Returns:
+        problem_definition (ProblemDefinition): the plaid problem definition initialized from the huggingface dataset description
+    """
+    problem_definition = ProblemDefinition()
+    problem_definition.set_task(description["task"])
+    problem_definition.set_split(description["split"])
+    problem_definition.add_input_scalars_names(description["in_scalars_names"])
+    problem_definition.add_output_scalars_names(description["out_scalars_names"])
+    problem_definition.add_input_timeseries_names(description["in_timeseries_names"])
+    problem_definition.add_output_timeseries_names(description["out_timeseries_names"])
+    problem_definition.add_input_fields_names(description["in_fields_names"])
+    problem_definition.add_output_fields_names(description["out_fields_names"])
+    problem_definition.add_input_meshes_names(description["in_meshes_names"])
+    problem_definition.add_output_meshes_names(description["out_meshes_names"])
+
+    return problem_definition
+
+
+class HFToPlaidSampleConverter:
+    """Class to convert a huggingface dataset sample to a plaid sample."""
+
+    def __init__(self, ds: datasets.Dataset):
+        self.ds = ds
+
+    def __call__(
+        self, sample_id: int
+    ):  # pragma: no cover  (not reported with multiprocessing)
+        """Convert a single sample from the huggingface dataset to a plaid sample."""
+        return Sample.model_validate(pickle.loads(self.ds[sample_id]["sample"]))
+
+
+class HFShardToPlaidSampleConverter:
+    """Class to convert a huggingface dataset sample shard to a plaid sample."""
+
+    def __init__(self, shard_path: Path):
+        self.ds = load_from_disk(shard_path.as_posix())
+
+    def __call__(
+        self, sample_id: int
+    ):  # pragma: no cover (not reported with multiprocessing)
+        """Convert a sample shard from the huggingface dataset to a plaid sample."""
+        sample = self.ds[sample_id]
+        return Sample.model_validate(pickle.loads(sample["sample"]))
```

Comment:
Are these classes supposed to be exposed to a user of the library or are they only defined as helper classes internally for your `huggingface_dataset_to_plaid` function ? If it's supposed to be hidden, then `_HFToPlaidSampleConverter` and `_HFShardToPlaidSampleConverter` could be approriate. Otherwise I think we need to populate the `__init__` and `__call__` docstrings